internal/controller/serving/reconcilers/kserve_keda_reconciler.go

```diff
@@ -0,0 +1,629 @@
+	}
+
+	expected.OwnerReferences = upsertOwnerReference(AsOwnerRef(isvc), curr)
+	expected.ResourceVersion = curr.ResourceVersion
+
+	if equality.Semantic.DeepDerivative(expected.Spec, curr.Spec) &&
+		equality.Semantic.DeepDerivative(expected.Labels, curr.Labels) &&
+		equality.Semantic.DeepDerivative(expected.Annotations, curr.Annotations) &&
+		equality.Semantic.DeepDerivative(expected.OwnerReferences, curr.OwnerReferences) {
+		log.V(1).Info("TriggerAuthentication is up-to-date")
+		return nil
+	}
+
+	log.Info("Updating TriggerAuthentication")
+	if err := k.client.Update(ctx, expected); err != nil {
+		return fmt.Errorf("failed to update TriggerAuthentication %s: %w", key.String(), err)
+	}
+	return nil
+}
+
+func (k *KserveKEDAReconciler) createTriggerAuthentication(ctx context.Context, isvc *kservev1beta1.InferenceService) error {
+	ta := k.expectedTriggerAuthentication(isvc)
+	if err := k.client.Create(ctx, ta); err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create TriggerAuthentication %s/%s: %w", ta.Namespace, ta.Name, err)
+	}
+	return nil
+}
+
+func (k *KserveKEDAReconciler) expectedTriggerAuthentication(isvc *kservev1beta1.InferenceService) *kedaapi.TriggerAuthentication {
+	return &kedaapi.TriggerAuthentication{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            KEDAPrometheusAuthTriggerAuthName,
+			Namespace:       isvc.Namespace,
+			OwnerReferences: []metav1.OwnerReference{AsOwnerRef(isvc)},
+			Labels:          getKedaLabels(isvc.Labels),
+			Annotations:     isvc.Annotations,
+		},
+		Spec: kedaapi.TriggerAuthenticationSpec{
+			SecretTargetRef: []kedaapi.AuthSecretTargetRef{
+				{
+					Parameter: "bearerToken",
+					Name:      KEDAPrometheusAuthTriggerSecretName,
+					Key:       "token",
+				},
+				{
+					Parameter: "ca",
+					Name:      KEDAPrometheusAuthTriggerSecretName,
+					Key:       "ca.crt",
+				},
+			},
+		},
+	}
+}
+
+// --- ServiceAccount ---
+func (k *KserveKEDAReconciler) reconcileServiceAccount(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+	expected := k.expectedServiceAccount(isvc)
+	curr := &corev1.ServiceAccount{}
+	key := client.ObjectKey{Namespace: expected.Namespace, Name: expected.Name}
+
+	err := k.client.Get(ctx, key, curr)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating ServiceAccount")
+			return k.createServiceAccount(ctx, isvc)
+		}
+		return fmt.Errorf("failed to get ServiceAccount %s: %w", key.String(), err)
+	}
+
+	expected.OwnerReferences = upsertOwnerReference(AsOwnerRef(isvc), curr)
+	expected.ResourceVersion = curr.ResourceVersion
+
+	if equality.Semantic.DeepDerivative(expected.Labels, curr.Labels) &&
+		equality.Semantic.DeepDerivative(expected.Annotations, curr.Annotations) &&
+		equality.Semantic.DeepDerivative(expected.OwnerReferences, curr.OwnerReferences) {
+		log.V(1).Info("ServiceAccount is up-to-date")
```

Comment:
I guess messages like this could be debug level.

Reply:
There is no debug level, maybe you mean level 2+

Reply:
Used level 2

Reply:
> There is no debug level, maybe you mean level 2+?

yes